vm/devices/tpm/src/ak_cert.rs

```diff
@@ -64,10 +63,21 @@ impl<T: 'static + RequestAkCert> From<T> for ResolvedRequestAkCert {
 /// A trait for requesting an AK cert.
 #[async_trait::async_trait]
 pub trait RequestAkCert: Send + Sync {
+    /// Helper function to create the request needed by `request_ak_cert`.
+    fn create_ak_cert_request(
```

Comment:
I'm not sure I understand why these traits were refactored the way they were? What assumption was broken from the previous implementation that required reworking these traits?

Reply:
Previously, we couple get report + request creation in a single call that works for isolated VM.
In more detail, the call does
1. Create an initial request
2. Create the hardware report that captures the hash of the subset of initial request
3. Finalize the request by adding the report to it

However, this contract does not work for trust launch VM that does not have the capability of getting hardware report.
So, this PR updates the contract into
1. Update the ` GetAttestationReport` to have `get_report` function for getting hardware report only
2. Add a `create_ak_cert_request` in `RequestAkCert` that takes the `Option<GetAttestationReport>`. For CVM, the function reserves the original behavior while for TVM, it skips the step 2 & 3.
3. The `request_ak_cert` in `RequestAkCert` now simply takes the request data as input and makes the GET call.